src/Uno.UI/UI/Xaml/Media/Imaging/BitmapImage.wasm.cs

```diff
@@ -0,0 +1,138 @@
+using Uno.Foundation;
+using Windows.Graphics.Display;
+
+using Path = global::System.IO.Path;
+
+namespace Windows.UI.Xaml.Media.Imaging
+{
+	public sealed partial class BitmapImage : BitmapSource
+	{
+		private static readonly string UNO_BOOTSTRAP_APP_BASE = Environment.GetEnvironmentVariable(nameof(UNO_BOOTSTRAP_APP_BASE));
+
+		private protected override bool TryOpenSourceAsync(int? targetWidth, int? targetHeight, out Task<ImageData> asyncImage)
+		{
+			var uri = WebUri;
+
+			if (uri != null)
+			{
+				var hasFileScheme = uri.IsAbsoluteUri && uri.Scheme == "file";
+				
+				// Local files are assumed as coming from the remote server
+				var newUri = hasFileScheme switch
+				{
+					true => new Uri(uri.PathAndQuery.TrimStart('/'), UriKind.Relative),
+					_	 => uri
+				};
+
+				asyncImage = AssetResolver.ResolveImage(newUri);
+				return true;
+			}
+
+			asyncImage = default;
+			return false;
+		}
+
+		internal static class AssetResolver
+		{
+			private static readonly Lazy<Task<HashSet<string>>> _assets = new Lazy<Task<HashSet<string>>>(() => GetAssets());
+
+			private static async Task<HashSet<string>> GetAssets()
+			{
+				var host = WebAssemblyRuntime.InvokeJS("window.location.host");
+				var scheme = WebAssemblyRuntime.InvokeJS("window.location.protocol");
+				var path = !string.IsNullOrEmpty(UNO_BOOTSTRAP_APP_BASE) ? $"{UNO_BOOTSTRAP_APP_BASE}/uno-assets.txt" : "uno-assets.txt";
+
+				var httpClient = new HttpClient();
+
+				var assets = await httpClient.GetStringAsync($"{scheme}//{host}/{path}");
+
+				return new HashSet<string>(Regex.Split(assets, "\r\n|\r|\n"));
+			}
+
+			public static async Task<ImageData> ResolveImage(Uri uri)
+			{
+				try
+				{
+					// ms-appx comes in as a relative path
+					if (uri.IsAbsoluteUri)
+					{
+						if (uri.Scheme == "http" || uri.Scheme == "https")
+						{
+							return new ImageData() { Kind = ImageDataKind.Url, Value = uri.AbsoluteUri };
+						}
+
+						// TODO: Implement ms-appdata
+						return new ImageData();
+					}
+					else
+					{
+						var assets = await _assets.Value;
+
+						return new ImageData() { Kind = ImageDataKind.Url, Value = GetScaledPath(uri.OriginalString, assets) };
+					}
+				}
+				catch (Exception e)
+				{
+					return new ImageData() { Kind = ImageDataKind.Error, Error = e };
+				}
+			}
+
+			private static string GetScaledPath(string path, HashSet<string> assets)
+			{
+				if (!string.IsNullOrEmpty(path))
+				{
+					var directory = Path.GetDirectoryName(path);
+					var filename = Path.GetFileNameWithoutExtension(path);
+					var extension = Path.GetExtension(path);
+
+					var resolutionScale = (int)DisplayInformation.GetForCurrentView().ResolutionScale;
+
+					for (var i = KnownScales.Length - 1; i >= 0; i--)
+					{
+						var probeScale = KnownScales[i];
+					
+						if (resolutionScale >= probeScale)
+						{
+							var filePath = Path.Combine(directory, $"{filename}.scale-{probeScale}{extension}");
```

Comment:
the image can set its scale like myimage.scale-100.png but also via the directory i.e. scale-100/myimage.png

Reply:
https://docs.microsoft.com/en-us/windows/uwp/app-resources/tailor-resources-lang-scale-contrast#reference-a-string-or-image-resource-by-name

Reply:
You are a 100% correct, this is the bare minimum we need to support for the gallery and will need to be revisited. This feature is significantly more complex than this : https://docs.microsoft.com/en-us/windows/uwp/app-resources/tailor-resources-lang-scale-contrast

This was agreed upon between Jérôme and François.